src/metabase/driver/sql_jdbc/connection.clj

```diff
@@ -166,7 +188,7 @@
   "Return an appropriate JDBC connection spec to test whether a set of connection details is valid (i.e., implementing
   `can-connect?`)."
   [driver details]
-  (let [details-with-tunnel (ssh/include-ssh-tunnel details)]
```

Comment:
this logic seems off to me. the `include-ssh-tunnel` and `incorporate-ssh-tunnel-details` each attempt to start an ssh tunnel. Seems like this is no good? If the tunnel is already open the `can-connect?` function will blow up on the tunnel's port already open?

Reply:
Yeah, although this point could have been made about the existing code as well.  In any case, it's updated in the latest review to only go through the "open" logic if required.